content/gsoc/2025/_index.md

```diff
@@ -3,10 +3,60 @@ title: PyBaMM GSoC 2025 Project Ideas
+
+### Technical details
+
+The implementation of the typing system will require careful evaluation of different approaches by the student, and they can explore either inline type hints in existing code, or separate `.pyi` stub files for backward compatibility, or the creation of a standalone `pybamm-stubs` package, or potentially a hybrid approach combining multiple methods. Each approach has its own trade-offs in terms of maintenance burden, backward compatibility, and ease of implementation – the student is expected to survey the existing strategy adopted by Scientific Python libraries and choose the most suitable approach for PyBaMM.
+
+Previously, type hints were added to the expression tree via [pybamm-team/pybamm#3578](https://github.com/pybamm-team/PyBaMM/issues/3578), which can serve as a reference for the student.
+
+The type system design will require particular attention to several key areas. The student may need to create custom types for battery-specific parameters, ensuring they accurately represent the domain concepts. The system must handle NumPy array types and dimensional analysis effectively, define clear type hierarchies for different battery models, and manage type compatibility with scientific computing libraries. The implementation of generic types for flexible model arguments will also prove to be essential for maintaining the PyBaMM framework's versatility.
+
+### Expected outcomes
+
+- Type system architecture: documentation of typing strategy and conventions, type hierarchy design for battery models, integration plan with existing codebase
+
+- Implementation: type hints for core pybamm.models API, custom type definitions for battery-specific components as needed, a short migration guide for adding types to other modules
+
+- Validation: CI integration with type checkers (Mypy, Pyright, basedmypy) as pre-commit hooks, documentation for type checking workflow and updates to the contributing guide
+
+- Documentation: Updated API documentation with type information, and a guide for downstream libraries on utilizing type information provided by PyBaMM
+
+### Desired skills
+
+- Some experience with type theory and static typing in Python is beneficial, but not required.
+- Python programming experience, Git version control, and GitHub workflow for open-source projects
+- An affinity for reading lots of code and documentation
+- An interest in scientific computing and battery modeling (prior experience not required)
+- As a plus, knowledge on how to use scientific computing libraries (NumPy, SciPy)
+- Some understanding of continuous integration providers (GitHub Actions, etc.) is beneficial, but not required.
+
+### Difficulty
+
+**Easy**. This project is suitable for a 175-hour duration.
+
+### Potential mentors
+
+TBA
```

Comment:
Happy to be here (google requires 1-2 mentors on each project at the time of submitting the application).

Reply:
Happy to add myself, too.